scipy/signal/waveforms.py

```diff
@@ -479,3 +480,70 @@ def _sweep_poly_phase(t, poly):
     intpoly = polyint(poly)
     phase = 2 * pi * polyval(intpoly, t)
     return phase
+
+
+def unit_impulse(shape, idx=None):
+    """
+    Unit impulse signal (discrete delta function) or unit basis vector.
+
+    Parameters
+    ----------
+    shape : int or tuple of int
+        Number of samples in the (1D) output, or a tuple that represents the
+        shape of the output.
+    idx : float or str, optional
```

Comment:
Should still list `None` as an option, especially since it's the default. And actually maybe it should be `'first'` instead, since we can make it more explicit during introspection that way?